scalableminds/webknossos
# Consistent serialization for SQL array literals

Fixes a bug where dataset tags with spaces were wrapped in quotes on reload (visible to the user).

Turns out Postgres array literals are weird.
* I now always add quotes when *writing* to string arrays to postgres. Note that when reading again, those quotes are only present if spaces are present too.
* If the user content contains quotes, we escape those with backslashes. We also unescape those when reading
* If the user content contains single quotes, we escape those by doubling them. We also unescape those when reading
* If the user content contains commas, we drop those. I could not figure out how to escape those

It may be worth updating slick or finding a better SQL connector where we do not have to write and parse those literals but can pass the values directly.

### URL of deployed dev instance (used for testing):
- https://escapetags.webknossos.xyz

### Steps to test:
- Add dataset tags with spaces, quotes and single quotes
- reload dataset list (reload twice to see updates!)
- should still look right

### Issues:
- fixes #6101 

------
